### PR TITLE
#12881 Implement Autoprinting PDF for Dunning 

### PR DESCRIPTION
**Pull request by cristinamghita**
#12881 Implement Autoprinting PDF for Dunning

### CODE REVIEW

File: backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/C_Async_Batch.java [hunk 1]
```diff
@@ -44,7 +44,8 @@
 	public void print(@NonNull final I_C_Async_Batch asyncBatch)
 	{
 		if (asyncBatch.isProcessed()
-				&& asyncBatchBL.isAsyncBatchTypeInternalName(asyncBatch, Async_Constants.C_Async_Batch_InternalName_InvoiceCandidate_Processing))
+				&& (asyncBatchBL.isAsyncBatchTypeInternalName(asyncBatch, Async_Constants.C_Async_Batch_InternalName_InvoiceCandidate_Processing)
```
Comment (metas-ts) [inline backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/C_Async_Batch.java:47]: better use the new two methods in `IAsyncBatchBL.java`

File: backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/ConcatenatePDFsCommand.java [hunk 2]
```diff
@@ -103,8 +108,8 @@ private void enqueuePrintQueues(@NonNull final PrintingQueueQueryRequest queryRe
 
 		final I_C_Async_Batch asyncBatch = asyncBatchBL.newAsyncBatch()
 				.setContext(ctx)
-				.setC_Async_Batch_Type(C_Async_Batch_InternalName_AutomaticallyInvoicePdfPrinting)
-				.setName(C_Async_Batch_InternalName_AutomaticallyInvoicePdfPrinting)
+				.setC_Async_Batch_Type(queryRequest.getAsyncBatchType())
+				.setName(queryRequest.getAsyncBatchType())
```
Comment (metas-ts) [inline backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/ConcatenatePDFsCommand.java:112]: have a user-friendly name using `AD_Message` (and the currentl user's language)

**Review by metas-ts — COMMENTED** [review]
minor requests

File: backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/ConcatenatePDFsCommand.java (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/ConcatenatePDFsCommand.java:112]: @metas-ts  (cc @LarsSander ) What should the name message contain?

Comment (metas-ts) [reply backend/de.metas.printing/de.metas.printing.base/src/main/java/de/metas/printing/model/validator/ConcatenatePDFsCommand.java:112]: @metas-rc @LarsSander i would say "Dunning Run" / "Mahnlauf"

**Pull request merged**